File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/Functions/SecondFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions
{
    public class SecondFunction : SingleArgumentFunction
    {
        public const string Name = "SECOND";

        protected override string FunctionHelp => $"'{Name} expects 1 arguments (Date)";

        protected override VariableType ArgumentType => PrimitiveType.Date;
        protected override VariableType ResultType => PrimitiveType.Number;

        private SecondFunction(Expression valueExpression, SourceReference reference)
            : base(valueExpression, reference)
        {
        }

        public static BuiltInFunction Create(SourceReference reference, Expression expression) =>
            new SecondFunction(expression, reference);
    }
}

File: typescipt/src/language/expressions/functions/secondFunction.ts


export class SecondFunction extends SingleArgumentFunction {
   public const string Name = `SECOND`;

   protected override string FunctionHelp => $`'{Name} expects 1 argument (Date)`;

   protected override VariableType ArgumentType => PrimitiveType.Date;
   protected override VariableType ResultType => PrimitiveType.Number;

   constructor(valueExpression: Expression, reference: SourceReference)
     : base(valueExpression, reference) {
   }

   public static create(reference: SourceReference, expression: Expression): ExpressionFunction {
     return new SecondFunction(expression, reference);
   }
}
